dummyserver/handlers.py

```diff
@@ -19,33 +20,43 @@
 
 
 class Response:
-    def __init__(self, body="", status="200 OK", headers=None):
+    def __init__(
+        self,
+        body: Union[str, bytes, Sequence[Union[str, bytes]]] = "",
+        status: str = "200 OK",
+        headers: Optional[Sequence[Tuple[str, Union[str, bytes]]]] = None,
+    ) -> None:
         self.body = body
         self.status = status
         self.headers = headers or [("Content-type", "text/plain")]
 
-    def __call__(self, request_handler):
+    def __call__(self, request_handler: RequestHandler) -> None:
         status, reason = self.status.split(" ", 1)
         request_handler.set_status(int(status), reason)
         for header, value in self.headers:
             request_handler.add_header(header, value)
 
+        if isinstance(self.body, str):
+            request_handler.write(self.body.encode())
+        elif isinstance(self.body, bytes):
+            request_handler.write(self.body)
         # chunked
-        if isinstance(self.body, list):
+        else:
             for item in self.body:
                 if not isinstance(item, bytes):
                     item = item.encode("utf8")
                 request_handler.write(item)
                 request_handler.flush()
-        else:
-            body = self.body
-            if not isinstance(body, bytes):
-                body = body.encode("utf8")
 
-            request_handler.write(body)
+
+RETRY_TEST_NAMES: Dict[str, int] = collections.defaultdict(int)
 
 
-RETRY_TEST_NAMES = collections.defaultdict(int)
+def request_params(request: httputil.HTTPServerRequest) -> Dict[str, bytes]:
+    params = {}
+    for k, v in request.arguments.items():
```

Comment:
This is changing where we get `request.params` to `request.arguments`, it looks like these arguments can come from both the querystring or the body, is that what we're intending here? (I can't find a reference to `HTTPServerRequest.params` in the Tornado docs)

Reply:
I should have explained this part. Previously the `_call_method` function monkey-patched the `params` attribute onto the request like so:

```py
        req.params = {}
        for k, v in req.arguments.items():
            req.params[k] = next(iter(v))
```

but mypy doesn't like this, so instead of assigning and using `request.params`, I use `params = request_params(request))` as needed.